dynatrace/environment_v1/oneagent.py

```diff
@@ -0,0 +1,131 @@
+"""
+Copyright 2021 Dynatrace LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+from enum import Enum
+from typing import List, Optional, Union
+from datetime import datetime
+
+from dynatrace.pagination import PaginatedList
+from dynatrace.dynatrace_object import DynatraceObject
+from dynatrace.http_client import HttpClient
+
+
+class MonitoringType(Enum):
+    CLOUD_INFRASTRUCTURE = "CLOUD_INFRASTRUCTURE"
+    FULL_STACK = "FULL_STACK"
+    STANDALONE = "STANDALONE"
+
+class AvailabilityState(Enum):
+    MONITORED = "MONITORED"
+    UNMONITORED = "UNMONITORED"
+    CRASHED = "CRASHED"
+    LOST = "LOST"
+    PRE_MONITORED = "PRE_MONITORED"
+    SHUTDOWN = "SHUTDOWN"
+    UNEXPECTED_SHUTDOWN = "UNEXPECTED_SHUTDOWN"
+    UNKNOWN = "UNKNOWN"
+
+
+class AutoUpdate(Enum):
+    ENABLED = "ENABLED"
+    DISABLED = "DISABLED"
+
+
+class OsType(Enum):
+    AIX = "AIX"
+    LINUX = "LINUX"
+    WINDOWS = "WINDOWS"
+    SOLARIS = "SOLARIS"
+    ZOS = "ZOS"
+
+    
+class OneAgentonaHostService:
+    def __init__(self, http_client: HttpClient):
+        self.__http_client = http_client
+
+
+    def list(
+        self, 
+        include_details: Optional[bool] = None, 
+        start_timestamp: Optional[Union[datetime, int]] = None,
```

Comment:
start_timestamp and end_timestamp must accept datetime, but when I pass a datetime this breaks.

You need to convert a datetime to int64 with `start_timestamp.timestamp() * 1000` if the parameter is datetime. I suggest you add a function `datetime_to_int64` to `utils.py`, that does the inverse of `int64_to_datetime`